### PR TITLE
BZ-1930354: Update yaml example for virt nmstate

### PR DESCRIPTION
**Pull request by lmandavi**
This PR is associated with https://bugzilla.redhat.com/show_bug.cgi?id=1930354.

### CODE REVIEW

**Comment by lmandavi** [comment]
@dphillip@redhat.com - Please review the updated content and provide your input.

**Comment by lmandavi** [comment]
The value "true" is present in the address portion of the yaml. Please view the attached image.
Screenshot from 2021-11-10 12-30-28

**Comment by lmandavi** [comment]
@dav1x - Please review the attached screenshot and my note.

**Comment by dav1x** [comment]
Ah! Awesome! Then LGTM!

**Comment by lmandavi** [comment]
@dav1x Thanks!
@myakove - Requesting QE review. If the content is approved, please add LGTM to this GitHub issue and move the associated defect to VERIFIED state.

**Comment by myakove** [comment]
@yossisegev can you please review?

**Review by yossisegev — CHANGES_REQUESTED** [review]
Two things:
1. Why should you override the DHCP case? Why not adding the static example, so the doc will include both static and dynamic setting?
2. The need to move the `enabled` field to be under `address` rather than under `ipv4` seems like a bug to me. This field is supposed to declare the will to enable the IPv4 capability of the interface.
I would like to further investigate it, and I'll get back to you with my findings. In the meantime - I can't approve the PR.

**Comment by dav1x** [comment]
@yossisegev 
I agree concerning the DHCP case. Both examples would be helpful.

The enabled field is under the the interface rather than IPv4 because there could be scenarios with multiple interfaces that need to be configured or maybe disable an interface and enable another one. 

nmstate examples show more of that:
https://github.com/nmstate/nmstate/blob/21faecd7211e5cd023868765468969324df5c8e2/tests/integration/static_ip_address_test.py#L202-L236

**Comment by yossisegev** [comment]
> The enabled field is under the the interface rather than IPv4 because there could be scenarios with multiple interfaces that need to be configured or maybe disable an interface and enable another one.

@dav1x There is no `enabled` field under the interface; what you have there is the `state` field, which can have the values `up/down/absent`, in order to serve exactly this purpose ("...maybe disable an interface and enable another one").
The `enabled` field belongs under the `ipv4` dict, and its purpose is to enable/disable IPv4 connectivity of the interface. The connectivity can be implemented either dynamically (in which case you'll have `dhcp: true`) or statically (in which you`ll have the `address` dict).
Anyway, in order to re-assure that, I have just applied a policy with the configuration suggested in this doc PR, i.e. `enabled` under `address`, and it failed.

File: modules/virt-example-nmstate-IP-management.adoc [hunk 1]
```diff
@@ -60,12 +60,14 @@ The following snippet configures an Ethernet interface that uses a dynamic IP ad
 ...
```
Comment (yossisegev) [inline modules/virt-example-nmstate-IP-management.adoc:60]: You changed the example to a static one, but left the intro in line 56 with "The following snippet.... that uses a *dynamic* IP address".

Comment (yossisegev) [inline modules/virt-example-nmstate-IP-management.adoc:60]: Please keep the dynamic example as it was, and add a static example as well.
In the static example - please keep `enabled` under `ipv4` and not under `address`.
Thank you!

**Review by yossisegev — CHANGES_REQUESTED** [review]

File: modules/virt-example-nmstate-IP-management.adoc (hunk 1 above)
Comment (dav1x) [reply modules/virt-example-nmstate-IP-management.adoc:60]: Oops. What I was saying and what I listed above are different. My apologies!

Comment (lmandavi) [reply modules/virt-example-nmstate-IP-management.adoc:60]: @yossisegev - To summarize your recommendations, is this what we want for this contet:
Examples: IP management
  Static
  No IP address
  Dynamic host configuration
  Static host configuration
  DNS
  Static routing

**Comment by yossisegev** [comment]
You can expand the doc to whatever you see fit. What I meant in my comments is :
1 Don't override the existing DHCP example with a new static IP example, but rather add the static IP example, so you'll have both the DHCP and the static IP examples.
2. Keep the `enabled` field under `ipv4`, rather than moving it to be under `address`.
As for the other caees (DNS, static routing, etc.) - I really can't answer that.

File: modules/virt-example-nmstate-IP-management.adoc (hunk 1 above)
Comment (lmandavi) [reply modules/virt-example-nmstate-IP-management.adoc]: Retained the content for dynamic example. Added the static example.

Comment (lmandavi) [reply modules/virt-example-nmstate-IP-management.adoc]: Updated the content.

**Comment by lmandavi** [comment]
@yossisegev - Please review the new topic on static IP example that was added to this section:
https://deploy-preview-38504--osdocs.netlify.app/openshift-enterprise/latest/virt/node_network/virt-updating-node-network-config.html#virt-example-nmstate-IP-management_virt-updating-node-network-config

**Comment by lmandavi** [comment]
@yossisegev - Please review the new topic on static IP example that was added to this section:
https://deploy-preview-38504--osdocs.netlify.app/openshift-enterprise/latest/virt/node_network/virt-updating-node-network-config.html#virt-example-nmstate-IP-management_virt-updating-node-network-config

**Comment by yossisegev** [comment]
@lmandavi I am repeating my last comments, regarding the new section `Static IP address host configuration`:
1.` prefix-length` is missing.
2. The new addition seems to be identical to the static section which is already covered in line 14 (https://deploy-preview-38504--osdocs.netlify.app/openshift-enterprise/latest/virt/node_network/virt-updating-node-network-config.html#virt-example-nmstate-IP-management-static_virt-updating-node-network-config), except for the missing `prefix-length`.

**Comment by lmandavi** [comment]
@yossisegev - Noted your comments. I did notice the similarity of the new topic and the existing topic on static IP. To avoid duplicity I can remove the new topic, and update the existing static section to include the prefix-length.

**Comment by yossisegev** [comment]
@lmandavi Sounds good to me!
Note that unless I am looking at a wrong section - the [existing static topic](https://deploy-preview-38504--osdocs.netlify.app/openshift-enterprise/latest/virt/node_network/virt-updating-node-network-config.html#virt-example-nmstate-IP-management-static_virt-updating-node-network-config) already has the `prefix-length` in the example, which looks valid.

**Comment by lmandavi** [comment]
@yossisegev - Right on! I see "prefix-length" in that existing static IP example. You were looking at the right content. Allow me to remove the redundant new topic added for static IP config and I will share the updated content with you shortly. Thanks!

**Comment by lmandavi** [comment]
@yossisegev - This is the latest HTML that I generated. The redundant example for static IP has been removed and the existing topic for Static Routing is retained. If this looks good, please approve the PR. Thank you.

https://deploy-preview-38504--osdocs.netlify.app/openshift-enterprise/latest/virt/node_network/virt-updating-node-network-config.html#virt-example-nmstate-IP-management_virt-updating-node-network-config

**Comment by yossisegev** [comment]
/LGTM
But I must ask - does this PR present any change, now that you removed the redundant section?
 I think it doesn't, unless I am missing something.

**Comment by lmandavi** [comment]
@yossisegev - Two things happened in this PR:
1) the example for dynamic host configuration was changed but the change was reversed after you tested and provided your feedback. So the original example for dynamic host configuration was retained.

2) I added a new example for static IP configuration, but you noted that the existing example in the doc is sufficient. To avoid duplication of content, I removed the new example that I added for static IP configuration.

To summarize, the original content for teh 4.9 release is retained, and no new changes are being implemented for this content.

I will update the Jira story and defect, and then close this PR without any merges.

**Comment by lmandavi** [comment]
Closing this PR because content changes were not implemented.
